creator-node/src/utils.js

```diff
@@ -101,6 +101,32 @@ const ipfsSingleByteCat = (path, logContext, timeout = 1000) => {
   })
 }
 
+/**
+ * Stat of a file at given filepath. If ipfs.files.stat() call takes longer than the timeout time or
+ * something goes wrong, an error will be thrown.
+*/
+const ipfsStat = (CID, logContext, timeout = 1000) => {
+  const logger = genericLogger.child(logContext)
+
+  return new Promise(async (resolve, reject) => {
+    const start = Date.now()
+    const timeoutRef = setTimeout(() => {
+      logger.error(`ipfsStat - Timeout`)
+      reject(new Error('IPFS Stat Timeout'))
+    }, timeout)
+
+    try {
+      const stats = await ipfsLatest.files.stat(`/ipfs/${CID}`)
```

Comment:
why use stat here instead of cat?

Reply:
to get the total size of the file -- and i think it accomplishes the same effect as cat (it will hang/timeout if the file isn't founds)